### PR TITLE
Draft of instructions for setting up Mattermost server using TLS

### PR DESCRIPTION
**Pull request by crspeller**

### CODE REVIEW

**Comment by esethna** [comment]
Thanks CS! I made some minor tweaks, +1

**Comment by esethna** [comment]
Do not merge until https://github.com/mattermost/platform/pull/4119 goes in

**Comment by esethna** [comment]
@crspeller can you please rebase this onto the `v3.5-documentation` branch so we can merge before release?

File: source/install/prod-ubuntu.rst [hunk 1]
```diff
@@ -303,6 +303,20 @@ Set up NGINX with SSL (Recommended)
   * Test the SSL certificate by visiting a site such as `https://www.ssllabs.com/ssltest/index.html <https://www.ssllabs.com/ssltest/index.html>`_
   * If there’s an error about the missing chain or certificate path, there is likely an intermediate certificate missing that needs to be included
 
+Setup HTTP2
+------------
+
+It is recomented to enable HTTP2 for inhanced performance.
```
Comment (jwilander) [inline source/install/prod-ubuntu.rst]: `inhanced` should be `enhanced`

File: source/install/prod-ubuntu.rst [hunk 2]
```diff
@@ -303,6 +303,20 @@ Set up NGINX with SSL (Recommended)
...
+It is recomented to enable HTTP2 for inhanced performance. 
+
+1. Modify your NGINX configuration as above. Then,
+
+  - Change the line ``listen 443 ssl;`` to ``listen 443 ssl http2;``
+  - Change the line ``proxy_pass http://10.10.10.2:8065;`` to ``proxy_pass https://10.10.10.2:8065;``
+  
+2. Restart NGINX
+
+3. Setup TLS on the mattermost server by following `these instrucions. <setup-tls>`_
```
Comment (jwilander) [inline source/install/prod-ubuntu.rst]: I think we're suppose to capitalize Mattermost everywhere, right?

File: source/install/setup-tls.rst [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+Setting up Mattermost server to run TLS
+========================================
+
+1. Go to the **General** > **Configuration** section of the System Console.
+2. Change the **Listen Address** setting to ``:443``
+3. Change the **Connection Security** setting to ``TLS``
+4. Change the **Forward port 80 to 443** setting to ``true`` if you wish to redirect users that try to connect insecurely to a secure connection. If you're using a proxy such as NGINX in front of Mattermost this setting is unnecessary and should be set to ``false``
+5. Run ``sudo setcap cap_net_bind_service=+ep ./bin/platform`` in your Mattermost directory to allow Mattermost to bind to low ports. You will need to re-run this command every time you upgrade Mattermost or it will fail to bind to the port.
+
+At this point you have two options. Automatic certificate retrieval though Let's Encrypt. Or manually specifying a certificate.
```
Comment (jwilander) [inline source/install/setup-tls.rst]:
```
At this point you have two options. Automatic certificate retrieval though Let's Encrypt. Or manually specifying a certificate.
```
should be a single sentence
```
At this point you have two options: automatic certificate retrieval though Let's Encrypt or manually specifying a certificate.
```

File: source/install/setup-tls.rst [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+
+Automatic Certificate Retreval
+===============================
+
+In this mode a certificate will be automatically retrieved the first time a client tries to connect to the Mattermost server. Certificates will be retrieved for any hostname a client tries to reach the server at. Setting this up is only one step:
+
+1. Change the **Use Let's Encrypt** setting to ``true``
```
Comment (jwilander) [inline source/install/setup-tls.rst]: Missing period at the end

File: source/install/setup-tls.rst [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+2. Restart the Mattermost server for these changes to take effect.
+
+
+Manual Certificate Specification
+=================================
+
+1. Change the **Use Let's Encrypt** setting to ``false``
```
Comment (jwilander) [inline source/install/setup-tls.rst]: Missing period at the end

File: source/install/prod-ubuntu.rst (hunk 2 above)
Comment (jwilander) [inline source/install/prod-ubuntu.rst]: Also random underscore at the end

**Comment by esethna** [comment]
Thanks! Tweaked and rebased in https://github.com/mattermost/docs/pull/608

**Pull request closed without merging**